src/interfaces.py

```diff
@@ -1,5 +1,105 @@
 from collections.abc import Callable
-from typing import Protocol
+from enum import Enum
+from typing import List, Protocol, Tuple
+
+
+class ISimulationObject(Protocol):
+    """An object synchronized with the simulation"""
+
+    traci_id: str
+
+    def update(self, updates: dict) -> None:
+        """Gets called whenever a simualtion tick has happened.
+
+        :param updates: The updated values for the synchronized properties
+        """
+
+    def add_subscriptions(self) -> int:
+        """Gets called when this object is created to allow
+        specification of simulator-synchronized properties.
+
+        :return: The synchronized properties (see <https://sumo.dlr.de/pydoc/traci.constants.html>)
+        """
+
+
+class ISimulationNode(ISimulationObject, Protocol):
+    """A point somewhere in the simulation where `ISImulationTrack`s meet"""
+
+    position: Tuple[float, float]
```

Comment:
@CR1337 What does our Protocol-expert says about properties in Protocols? 🤔

Reply:
> Does users of the interface have access to the properties? thinking Otherwise it should be part of the interface, I think, because the user shouldn't know the concrete functionality. innocent But if so, then please ad the functions with the `@property`-tag Sorry for the spongy definition of the term "interface" disappointed

In my opinion any element (attribute, method, property, ...) should be part of the protocol when an outside object should be able to access it.

Reply:
> See https://peps.python.org/pep-0544/#protocol-members This might have to be changed anyway, as some properties should be read-only.

Properties are always readonly as long as you only specify a `@property`-decorator. You have to make it writable explicitly. For example for a property `test` you would have to define another method decorated with `@test.setter`.

Reply:
Good to know 👍 Thanks for the clarification! I learn something new every day.